snapcraft/internal/parser.py

```diff
@@ -267,15 +274,15 @@ def _process_index(output):
     for line in output.decode().splitlines():
         if line == '---':
             if entry:
-                _process_wiki_entry(entry, master_parts_list)
+                errors['wiki'] += _process_wiki_entry(entry, master_parts_list)
                 entry = ''
         else:
             entry = '\n'.join([entry, line])
 
     if entry:
-        _process_wiki_entry(entry, master_parts_list)
+        errors['wiki'] += _process_wiki_entry(entry, master_parts_list)
 
-    return master_parts_list
+    return (master_parts_list, errors)
```

Comment:
I like that we are returning the number of errors, but in python we shouldn't return a tuple like in go.
Instead, if errors >  0, raise an exception. You can add an attribute to the WikiError class that's number_of_errors, so you can extract it later.